src/main/scala/com/microsoft/hyperspace/index/rules/FilterIndexRule.scala

```diff
@@ -109,7 +109,7 @@ object FilterIndexRule
         val newRelation = HadoopFsRelation(
           newLocation,
           new StructType(),
-          index.schema,
+          StructType(index.schema.filter(fsRelation.schema.contains)),
```

Comment:
Question. If we just use `fsRelation.schema` here instead of `index.schem`, it would still work since we already verified that all the columns being referenced are covered?